project/event_logging.py

```
@@ -0,0 +1,278 @@
+class EventLoggingRoute(APIRoute):
+    """Route class to wrap route handler to log events when endpoints are called."""
+
+    def get_route_handler(self):
+        original_handler = super().get_route_handler()
+
+        # TODO: handle exceptions for streaming responses
+        async def log_event(request: Request):
+            try:
+                response = await original_handler(request)
+                event_logger.log_event(request, response.status_code)
+                return response
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)